cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -552,7 +552,7 @@ class SsaPhiNode extends Node, TSsaPhiNode {
    */
   final Node getAnInput(boolean fromBackEdge) {
     localFlowStep(result, this) and
-    if phi.getBasicBlock().dominates(result.getBasicBlock())
+    if phi.getBasicBlock().strictlyDominates(result.getBasicBlock())
```

Comment:
It's no clear to me why this change is correct. It might be the right thing to do for the invalid pointer deref query, but the `getAnInput` predicate is also used in other contexts.

Reply:
FWIW, the only use other than in the `cpp/invalid-pointer-deref` query is in https://github.com/github/codeql/blob/733a00039efdb39c3dd76ddffad5e6d6c85e6774/cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll#L997. I tried to run it on the problematic projects that made us introduce the usage there, and that stills works out fine.

Reply:
If we don't want to change the behavior in the library, we could alternatively add an `phi != node` in the [introduced barrier](https://github.com/github/codeql/pull/12971/files#diff-f245ae0a243ae41fb7d507426991139388edd94c3dee54575550955762664b3cR233)? That should have the same effect.